### ISSUE TITLE
Improve Screen Reader support for Angular Modals

### ISSUE THREAD

**Issue by bobwise** [sparkdesignsystem/spark-design-system#1652]
## Describe The Bug
There are two items of functionality that will improve screen reader support for Angular Modals:

1. Cursor Trapping. When the modal is opened, the focus cursor (and the VoiceOver cursor) are moved into the modal and trapped there.  The cursor cycles between the elements inside the modal and elements outside the modal cannot be focused. In the Choice variant, the confirm button is focused by default. In the Info variant, the Close icon is focused by default. In the Wait variant, the entire modal is focused by default.
2. Use `aria-describedby` and `aria-labeledby` so that the screen reader correctly identifies the label and content of the modal when it is opened.
